dev_tools/utils/nxdl_utils.py

```diff
@@ -39,8 +30,35 @@ def get_nexus_definitions_path():
     try:  # either given by sys env
         return os.environ["NEXUS_DEF_PATH"]
     except KeyError:  # or it should be available locally under the dir 'definitions'
-        local_dir = os.path.abspath(os.path.dirname(__file__))
-        return os.path.join(local_dir, f"..{os.sep}..")
+        local_dir = Path(__file__).resolve().parent
+        for _ in range(2):
+            local_dir = local_dir.parent
+        return local_dir
+
+
+nexus_def_path = get_nexus_definitions_path()
+
+
+def get_app_defs_names():
+    """Returns all the AppDef names without their extension: .nxdl.xml"""
+    app_def_path_glob = nexus_def_path / "applications" / "*.nxdl*"
```

Comment:
No, these are `Path` objects, so using `/` concatenates the string according to the platform (e.g., `\` for windows and `/` for unix based systems). So it should be preferred over f-strings, because it handles this more sensibly than using a fixed separator.